### DISCUSSION TITLE
Feature Request: Add functionality to dynamically change vector store metadata filters

### DISCUSSION THREAD

**Discussion by BennyEV7** [langflow-ai/langflow#2878] [Ideas]
Hi everyone, is there any way to dynamically update the value field of the "Search Metadata Filter" for the Astra DB vector store? Ideally (I think, maybe?) I could add a connector that passes the value(s) to this field. 

I'm currently using a prompt that checks the chat input against a list of client names and returns only the client names contained in the chat input. I would like the ability to pass these client names to the metadata filter before the vector search process initiates.  

See the attached screenshot for an example of how I'm imagining this would flow.

I'm all ears if there is a better way to filter the vector store results.
![Screenshot 2024-07-22 142221](https://github.com/user-attachments/assets/c15d018b-59b0-4ecc-a0f0-5636d0a507ef)

**Comment by jordanrfrazier** [comment]
I think this is a great idea. I'll speak with the team and get this prioritized. Thanks for the submission!

**Reply by dougzec** [reply]
@jordanrfrazier What is the rationale behind removing this sort of filtering in the more recent versions?

**Reply by jordanrfrazier** [reply]
@erichare Can you comment on this?

**Reply by erichare** [reply]
<img width="1675" height="114" alt="Screenshot 2025-08-13 at 8 05 20 AM" src="https://github.com/user-attachments/assets/0da51fc2-a47b-4d68-9dc4-efe4cfd026e9" />

Hey all! so, see the attached screenshot ... in the advanced controls it is still there, but because filters can be more complex than just key value pairs, it was turned into a NestedDictInput - basically so the filters can be specified at any level of granularity.

It's definitely less user friendly for simple examples, but the functionality discussed here can be achieved by returning output in this format still. If there's a clean way we can be both user friendly but also allow more advanced filter parameters to be provided that would be fantastic of course.

**Comment by grahamHeathcote** [comment]
I had this same issue and made my own version of the AstraDB component.

Here is the link if it helps anyone in the future :)

https://github.com/grahamHeathcote/AstraDBComponentLangflow

**Reply by erichare** [reply]
Thanks for this!! it looks great, i plan to play with it more soon. If there's pieces of it that you think could be a nice addition to the existing component, feel free to open a PR and tag me, we'll triage it as a team
